### PR TITLE
Start tracking release build information in an azure storage table

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

Start tracking release build information in an azure storage table
  * Also, fix some issues found when the build is started through a schedule 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.  
- **User-facing changes**
    - [x] Not Applicable
    - **OR**  
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: tools/releaseBuild/azureDevOps/templates/updateBuildTable.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+jobs:
+- job: updateBuildTable
+  displayName: Update Build Information Tbale
```
Comment (adityapatwardhan) [inline tools/releaseBuild/azureDevOps/templates/updateBuildTable.yml]: Tbale -> Table

**Review by PaulHigin — APPROVED** [review]

File: tools/releaseBuild/azureDevOps/templates/updateBuildTable.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  dependsOn:
+    ${{ parameters.parentJobs }}
+  
+  # the package EZ pools don't have the latest module needed to run this job
```
Comment (adityapatwardhan) [inline tools/releaseBuild/azureDevOps/templates/updateBuildTable.yml]: package `ES`

File: tools/releaseBuild/azureDevOps/templates/updateBuildTable.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  pool:
+    name: Hosted Mac Internal
```
Comment (adityapatwardhan) [inline tools/releaseBuild/azureDevOps/templates/updateBuildTable.yml]: Any specific reason for this being macOS?

File: tools/releaseBuild/azureDevOps/templates/updateBuildTable.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+      $table = Get-AzStorageTableTable -resourceGroup pscoredata -storageAccountName pscorebuilddata -tableName buildData
+      
+      Write-Verbose "Removing existing row, if it exists ..." -verbose
+      $row = Remove-AzStorageTableRow -table $table -partitionkey '$(System.DefinitionId)' -rowKey '$(ReleaseTagVar)'
+
+      Write-Verbose "Adding new row ..." -verbose
+      $result = Add-AzStorageTableRow -table $table -partitionkey '$(System.DefinitionId)' -rowKey '$(ReleaseTagVar)' -property @{
+              buildid = '$(Build.BuildId)'
+
```
Comment (adityapatwardhan) [inline tools/releaseBuild/azureDevOps/templates/updateBuildTable.yml]: Extra line.

File: tools/releaseBuild/azureDevOps/templates/updateBuildTable.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+              buildid = '$(Build.BuildId)'
+      
+            }
+      While(!$result.IsCompleted)
+      {
+        Write-Verbose -message "waiting for result" -verbose
+        Start-Sleep -Seconds 1
+      }
+
+      if(!$result.IsCompletedSuccessfully)
```
Comment (adityapatwardhan) [inline tools/releaseBuild/azureDevOps/templates/updateBuildTable.yml]: Do we need a timeout?

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: tools/releaseBuild/azureDevOps/templates/updateBuildTable.yml (hunk 3 above)
Comment (TravisEz13) [reply tools/releaseBuild/azureDevOps/templates/updateBuildTable.yml]: Other than the comment, it's an internal pool

File: tools/releaseBuild/azureDevOps/templates/updateBuildTable.yml (hunk 5 above)
Comment (TravisEz13) [reply tools/releaseBuild/azureDevOps/templates/updateBuildTable.yml]: The whole job has a timeout, which should catch before compliance finishes.

I'll lowe it to 15 minutes.

**Comment by TravisEz13** [comment]
@adityapatwardhan 	 can you update your review?

**Comment by adityapatwardhan** [comment]
Can you have a look at the codefactor issues, it is reporting a few trailing spaces.

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**